Source/SwiftLintBuiltInRules/Rules/Idiomatic/StaticOverFinalClassRule.swift

```diff
@@ -0,0 +1,136 @@
+import SwiftSyntax
+
+@SwiftSyntaxRule
+struct StaticOverFinalClassRule: Rule {
+    var configuration = SeverityConfiguration<Self>(.warning)
+
+    static let description = RuleDescription(
+        identifier: "static_over_final_class",
+        name: "Static Over Final Class",
+        description: "Prefer `static` over `final class` for non-overridable declarations",
+        kind: .idiomatic,
+        nonTriggeringExamples: [
+            Example("""
+            class C {
+                static func f() {}
+            }
+            """),
+            Example("""
+            class C {
+                static var i: Int { 0 }
+            }
+            """),
+            Example("""
+            class C {
+                static subscript(_: Int) -> Int { 0 }
+            }
+            """),
+            Example("""
+            class C {
+                class func f() {}
+            }
+            """),
+            Example("""
+            final class C {}
+            """),
+            Example("""
+            final class C {
+                class D {
+                  class func f() {}
+                }
+            }
+            """)
+        ],
+        triggeringExamples: [
+            Example("""
+            class C {
+                ↓final class func f() {}
+            }
+            """),
+            Example("""
+            class C {
+                ↓final class var i: Int { 0 }
+            }
+            """),
+            Example("""
+            class C {
+                ↓final class subscript(_: Int) -> Int { 0 }
+            }
+            """),
+            Example("""
+            final class C {
+                ↓class func f() {}
+            }
+            """),
+            Example("""
+            class C {
+                final class D {
+                    ↓class func f() {}
+                }
+            }
+            """)
+        ]
+    )
+}
+
+// Stack of flags indicating whether each level in the tree is a final class
+private typealias LevelIsFinalClassStack = Stack<Bool>
+
+private extension LevelIsFinalClassStack {
+    var lastIsFinalClass: Bool { peek() == true }
+}
+
+private extension StaticOverFinalClassRule {
+    final class Visitor: ViolationsSyntaxVisitor<ConfigurationType> {
+        private var levels = LevelIsFinalClassStack()
```

Comment:
I think this can be simplified:
```suggestion
        private var classContexts = Stack<Bool>
```